src/test/java/org/opensearch/neuralsearch/processor/ScoreNormalizationIT.java

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor;
+
+import static org.opensearch.neuralsearch.TestUtils.assertHybridSearchResults;
+import static org.opensearch.neuralsearch.TestUtils.createRandomVector;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+
+import lombok.SneakyThrows;
+
+import org.junit.After;
+import org.junit.Before;
+import org.opensearch.index.query.QueryBuilders;
+import org.opensearch.knn.index.SpaceType;
+import org.opensearch.neuralsearch.common.BaseNeuralSearchIT;
+import org.opensearch.neuralsearch.query.HybridQueryBuilder;
+import org.opensearch.neuralsearch.query.NeuralQueryBuilder;
+
+import com.google.common.primitives.Floats;
+
+public class ScoreNormalizationIT extends BaseNeuralSearchIT {
```

Comment:
No need to change but just want to share my thought. Integration test is expensive. Therefore, it would be better to write IT just for the base functionality and leave all the other cases to unit test.
In this case, I would write IT using either L2 for min-max only as except the calculation logic, all others things are same between the two. The calculation logic can be covered in unit test.